lib/Doctrine/ORM/Query.php

```diff
@@ -403,10 +406,17 @@ private function processParameterMappings($paramMappings)
                 $value = array_keys(HierarchyDiscriminatorResolver::resolveDiscriminatorsForClass($value, $this->em));
             }
 
+            if (is_object($value) && $this->em->getMetadataFactory()->hasMetadataFor(get_class($value))) {
+                $metadata = $this->em->getClassMetadata(get_class($value));
+                $type     = $metadata->getColumn($metadata->getIdentifier()[0])->getTypeName();
```

Comment:
What happens when the `$value` is a composite-identified entity? Should we throw?